### ISSUE TITLE
Import sorting

### ISSUE THREAD

**Issue by ewhauser** [aspect-build/bazel-super-formatter#20]
A few languages have some (relatively) standard tools for sorting imports (https://github.com/PyCQA/isort, https://pkg.go.dev/golang.org/x/tools/cmd/goimports) that are separate from the formatters. Is there openness to adding those as a part of super formatter or considered out of scope?

**Comment by alexeagle** [comment]
isort is breaking, because python imports can (and often do) have side-effects. Per the design doc, I'd rather this tool only make edits that developers can blindly accept.
(https://hackmd.io/0UgIb6gyTvSVX9N2vTPGug)

goimports feels like it might qualify as a formatter per that table, though. I don't think go imports can have side-effects... do you think so?

**Comment by ewhauser** [comment]
Good point and makes sense on isort.

Technically, I think there can be side effects in Go if you are using `init()` (https://go.dev/ref/spec#Package_initialization) and relying on order? I've never seen this behavior relied on in practice in a Go app though. 

```
If a package has imports, the imported packages are initialized before initializing the package itself. If multiple packages import a package, the imported package will be initialized only once. The importing of packages, by construction, guarantees that there can be no cyclic initialization dependencies.
```

A more experienced Go developer might have better thoughts here.

**Comment by alexeagle** [comment]
Seems to me like we need good linting support anyway, so we should default to having these things there if it's not clear they are safe to apply blindly.

If you have some cycles, I'd be happy to walk through what I imagine are next steps for linting!

**Comment by ewhauser** [comment]
Sure; let me know what you are thinking.

**Comment by alexeagle** [comment]
okay let's sync on slack, I think this one is probably not a formatting job.
